From 570419da380733487284e715ab55c3a5638e98ad Mon Sep 17 00:00:00 2001
From: b38613 <B38613@freescale.com>
Date: Tue, 15 Nov 2011 16:16:39 +0800
Subject: [PATCH 210/635] ENGR00161427 GPS: UART has too many interrupts

disable service in init.rc and fix sometimes GPS init error.

Signed-off-by: b38613 <B38613@freescale.com>
---
 libgps/athr_gps.c | 17 +++++++++++++----
 1 file changed, 13 insertions(+), 4 deletions(-)

diff --git a/libgps/athr_gps.c b/libgps/athr_gps.c
index 3b1b4a7..fda3be0 100755
--- a/libgps/athr_gps.c
+++ b/libgps/athr_gps.c
@@ -1497,18 +1497,23 @@ Fail:
 /*****************************************************************/
 /*****************************************************************/
 
-
 static int athr_gps_init(GpsCallbacks* callbacks)
 {
     GpsState*  s = _gps_state;
 
     s->callbacks = *callbacks;
+    if (property_set("ctl.start", "ing") < 0) {
+        LOGE("Failed to start ing");
+        return -1;
+    }
 
-    if (!s->init)
-        gps_state_init(s);
+    gps_state_init(s);
 
     if (s->fd < 0)
+      {
+	LOGE("athr_init error\r\n");
         return -1;
+      }
 
     return 0;
 }
@@ -1520,6 +1525,10 @@ athr_gps_cleanup(void)
 
     if (s->init)
         gps_state_done(s);
+    if (property_set("ctl.stop", "ing") < 0) {
+      LOGE("Failed to stop ing");
+      return;
+    }
 }
 
 static int athr_run_hook(char* name)
@@ -1655,7 +1664,7 @@ static int athr_gps_set_position_mode(GpsPositionMode mode, GpsPositionRecurrenc
 
 	if (!s->init || min_interval < 0) {
 		D("%s: called with uninitialized state !!", __FUNCTION__);
-		return -1;
+		//return -1;
 	}
 
 	s->fix_freq = min_interval/1000;
-- 
1.8.0

